### PR TITLE
changes the networking to calico

### PR DESCRIPTION
**Pull request by twmobius**

### CODE REVIEW

File: templates/k0s.yaml [hunk 1]
```diff
@@ -40,7 +40,15 @@ spec:
     adminPort: 8133
     agentPort: 8132
   network:
-    calico: null
+    calico:
+      mode: vxlan
+      vxlanPort: 4789
+      vxlanVNI: 4096
+      mtu: 1450
```
Comment (akosiaris) [inline templates/k0s.yaml:47]: why not set it to 0 and rely on autodetection?

File: templates/k0s.yaml [hunk 2]
```diff
@@ -40,7 +40,15 @@ spec:
...
+      wireguard: false
+      flexVolumeDriverPath: /usr/libexec/k0s/kubelet-plugins/volume/exec/nodeagent~uds
```
Comment (akosiaris) [inline templates/k0s.yaml:49]: This, alongside most of the other settings (mtu aside) are the defaults. Do we need to define all of them? Or can just rely on the defaults?

**Review by akosiaris — COMMENTED** [review]
I think we lack a ferm rule for vxlan UDP port 4789 between the nodes

File: templates/k0s.yaml (hunk 2 above)
Comment (twmobius) [reply templates/k0s.yaml:49]: From what I get, the configuration applied via the k0sctl and by extension the k0s terraform resource, is being merged with the default for any values missing. (You can verify it by getting the running k0s.conf from the k0s backup command). However I think that specifying a setting explicitly saves up down the line if a default value changes during an upgrade.

**Comment by twmobius** [comment]
With regards to the ferm rule I think currently calico opens the ports up allowing access from anywhere, I guess this should be tighten via the calico configuration?

```
ACCEPT     udp  --  worker3.example.com  anywhere             udp dpt:4789
ACCEPT     udp  --  controller0.example.com  anywhere             udp dpt:4789
ACCEPT     udp  --  worker1.example.com  anywhere             udp dpt:4789
ACCEPT     udp  --  controller1.example.com  anywhere             udp dpt:4789
ACCEPT     udp  --  worker0.example.com  anywhere             udp dpt:4789
ACCEPT     udp  --  worker2.example.com  anywhere             udp dpt:4789
ACCEPT     udp  --  controller2.example.com  anywhere             udp dpt:4789
```

File: templates/k0s.yaml (hunk 2 above)
Comment (akosiaris) [reply templates/k0s.yaml:49]: Generally, upgrading without reading tons of changelogs isn't a good practice and default changes are red flags. But ok, hardcoding won't hurt either. But there is one default, MTU, where the default value, 0, does autodetection of MTU

**Comment by akosiaris** [comment]
> With regards to the ferm rule currently calico sets up iptables rules directly, I guess this should be tighten via the calico configuration?
> 
> ```
> ACCEPT     udp  --  worker3.example.com  anywhere             udp dpt:4789
> ACCEPT     udp  --  controller0.example.com  anywhere             udp dpt:4789
> ACCEPT     udp  --  worker1.example.com  anywhere             udp dpt:4789
> ACCEPT     udp  --  controller1.example.com  anywhere             udp dpt:4789
> ACCEPT     udp  --  worker0.example.com  anywhere             udp dpt:4789
> ACCEPT     udp  --  worker2.example.com  anywhere             udp dpt:4789
> ACCEPT     udp  --  controller2.example.com  anywhere             udp dpt:4789
> ```
> 
> Edit: Documentation mentions that if firewall is needed we should use the HostEndpoint and the GlobalNetworkPolicy (https://docs.tigera.io/calico-enterprise/3.15/getting-started/install-on-clusters/requirements)

Yeah, that one will need some hardening for sure.

**Comment by akosiaris** [comment]
I 'll merge to test and perhaps change the MTU default to 0.

**Comment by akosiaris** [comment]
16b86379 added the 4789 rules and it's by design. No hardening needed on that front.

**Pull request merged**